src/core/query-builder.utils.ts

```diff
@@ -71,6 +75,69 @@ function transformBasedOnContainsOperations(query: string, field: string, transf
     });
 }
 
+/**
+ * The function will replace inner query with Any operation with their transformation.
+ * This will handle @see QueryBuilderOperations.LIST_CONTAINS and @see QueryBuilderOperations.LIST_DOES_NOT_CONTAIN operations.
+ *
+ * for example:
+ * Input: keywords.Any(it.Contains({ "key1", "key2" }))
+ * Output: keywords.Any(it.Contains("key1") || it.Contains("key2"))
+ * 
+ * Input: keywords.Any(!it.Contains({ "key1", "key2" }))
+ * Output: !keywords.Any(it.Contains("key1") || it.Contains("key2"))
+ * 
+ * @param query Query string containing one or more instances of the Any operation to be transformed.
+ * @param field Name of the field on which the Any operation is performed.
+ * @param transformation callback function that transforms the value based on the operation.
+ * @param options The options to be used in the transformation
+ * @returns Transformed query string with Any operations processed.
+ */
+function transformBasedOnAnyOperations(
+  query: string,
+  field: string,
+  transformation: ExpressionTransformFunction,
+  options?: Map<string, Map<string, unknown>>
+) {
+  const anyRegex = new RegExp(`\\b${field}\\.Any\\s*\\((.*)\\)`, 'g');
+
+  return query.replace(anyRegex, (_match, innerPredicate: string) => {
+    const containsRegex = new RegExp(String.raw`(?:!(it\.Contains\("([^"]*)"\))|(it\.Contains\("([^"]*)"\)))`, 'g');
```

Comment:
There isn't any dynamic replacement in the regex here right? Can you just use normal inline regex syntax?

```ts
    const containsRegex = /(?:!(it\.Contains\("([^"]*)"\))|(it\.Contains\("([^"]*)"\)))/g;
```